src/views/login.jsx

```diff
@@ -1,7 +1,162 @@
-import React from "react";
+/* eslint-disable react/no-unescaped-entities */
+import { useState } from "react";
+import {
+  signInWithEmailAndPassword,
+  GoogleAuthProvider,
+  signInWithPopup,
+} from "firebase/auth";
+import { auth } from "../../firebase_config";
+import { Link, useNavigate } from "react-router-dom";
 
-const login = () => {
-  return <div>login</div>;
-};
+function Login() {
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const provider = new GoogleAuthProvider();
+  const navigate = useNavigate();
 
-export default login;
+  const handleLogIn = async (e) => {
+    e.preventDefault();
+
+    signInWithEmailAndPassword(auth, email, password)
+      .then(() => {
+        navigate("/dashboard");
+      })
+      .catch((error) => {
+        if (error.code === "auth/user-not-found") {
+          alert("user not found");
+        }
+        if (error.code === "auth/wrong-password") {
+          alert("your password is wrong");
+        }
+        if (error.code === "auth/invalid-email") {
+          alert("invalid email");
+        }
+        console.log(error);
+      });
+  };
+
+  const handleGoogleSignUp = async () => {
+    signInWithPopup(auth, provider)
+      .then(() => {
+        navigate("/dashboard");
+      })
+      .catch((error) => {
+        console.log(error);
+      });
+  };
+
+  return (
+    <div className="flex min-h-full flex-1 flex-col justify-center py-12 sm:px-6 lg:px-8">
+      <div className="sm:mx-auto sm:w-full sm:max-w-md">
+        <Link to="/">
+          <img
+            className="mx-auto h-28 w-auto"
+            src="https://res.cloudinary.com/yilin1234/image/upload/v1692498189/f__2_-removebg-preview_xxhyv5.png"
```

Comment:
we can store the image urls in a constants file since they are reused in other parts of the application.